Let there be (held) light! Adds directional flashlight sprites, and many new inhand icons for various light objects

vinylspiders opened the pull request:
## About The Pull Request

<details>
<summary>Directional flashlights are here!</summary>

![image](https://user-images.githubusercontent.com/13398309/229491332-5902aa4d-ff6b-4c16-80e2-9ae66183604d.png)

</details>

<details>
<summary>Has this ever bothered you? The light cone overlay is directional, but not the flashlight itself</summary>

![dreamseeker_zcnBNnAXKy](https://user-images.githubusercontent.com/13398309/229491045-de68f57e-9b24-43e9-99fa-508ac5140e84.gif)

</details>

Me too, so I went and made directional sprites for all of the flashlights that have directional lights. These flashlights change their direction when dropped or thrown to match the user doing the dropping. 


<details>
<summary>Shown here</summary>

![dreamseeker_XqMbjg0xMI](https://user-images.githubusercontent.com/13398309/229490937-e923970a-e94d-4502-b76c-8d597dc00730.gif)

</details>

### But wait, there's more!

---

I also added a number of on/off inhand sprites for _most_ of the holdable existing light sources in the game (sorry, making a tiny banana lamp will have to wait for another time).

<details>
<summary>Noteworthy ones: animated candles! and flares too</summary>

![dreamseeker_VEmAOfAXOm](https://user-images.githubusercontent.com/13398309/229493049-284925a0-b330-40ea-956e-090dad9ef251.gif)

![dreamseeker_LYWbSYv8la](https://user-images.githubusercontent.com/13398309/229942307-6dfbf28a-00a2-4abf-8ebd-dc1531f5e98e.gif)

</details>

---

### Other things

You can now light candles with another candle. Why was this not possible?

![dreamseeker_OJKsrqoVHT](https://user-images.githubusercontent.com/13398309/229491696-52bfbd9c-0807-42e8-9f54-25f79c17fc7c.gif)

<details>
<summary>More sprite makeovers/additions here</summary>

Desk lamp:

![dreamseeker_xnEp3V6D3C](https://user-images.githubusercontent.com/13398309/229492350-178d2765-ee61-4605-bf58-7e7a28a4b125.gif)

The iconic green lamp gets a makeover:

![dreamseeker_FYPhsdQ1TZ](https://user-images.githubusercontent.com/13398309/229492428-bbe7831e-8655-482d-a56a-a97f150d382d.gif)

_The_ flashlight:

![dreamseeker_bk9YRsxrl8](https://user-images.githubusercontent.com/13398309/229493976-d007edcc-002c-4803-a581-be864e80939b.gif)

Syndicate lantern:

![dreamseeker_Ypj1i7qwLs](https://user-images.githubusercontent.com/13398309/229492237-69c8d13a-a267-4532-81be-73cdd5dab7e4.gif)

Flashdark:

![dreamseeker_XoC0BnevAF](https://user-images.githubusercontent.com/13398309/229494042-97f9598c-80a0-4beb-b73b-163518d27a71.gif)

Lantern:

![dreamseeker_LSSuy7qYNJ](https://user-images.githubusercontent.com/13398309/229492283-aabd5bd4-1083-4196-a821-e9ffeac0f7bc.gif)

Wisp lantern (both a new inhand and icon sprite):

![dreamseeker_VhUyyjQMHv](https://user-images.githubusercontent.com/13398309/229493378-577f3878-2bb1-4ad5-91e4-05cd6ca69f52.gif)

Seclight:

![dreamseeker_4emki7TfOf](https://user-images.githubusercontent.com/13398309/229496047-662dcc57-7d81-4263-90e3-38f99e0a2eae.gif)

</details>

## Why It's Good For The Game

Better visuals for the game. Handheld lighting objects are probably some of the more commonly used items, and so having sprites that accurately represent them in game is a nice touch. 

Also did some code refactoring, cleanup/slight optimization tweaks while I was in the flashlight file. Notably candles were needlessly calling update_appearance every process tick. Now they will only do that if they actually need an update (if their wax level changes).

## Changelog
:cl:
fix: candles can now be used to light other candles, cigarettes, and anything else that needs lighting in a pinch.
fix: flashlights that have directional lights now have directional sprites to match
fix: candles can now be snuffed again
imageadd: adds new inhand sprites for most flashlights in the game, including animated flares and candles
refactor: cleaned up flashlight.dm's unnecessary bits and made some slight improvements
/:cl:

vinylspiders committed:
Adds directional sprites for flashlights

Beginning work on inhand icons.

vinylspiders committed:
Adds a whole bunch of worn icon states for flashlights, candles, lanterns.

You... not take... candle...

vinylspiders committed:
adds ability to light candles with other candles, and new animated flare inhands

vinylspiders committed:
quality pass 1

vinylspiders committed:
jesus saves

quality pass two, where I realize I lost the directional flashlight icons somehow and pulled them back from a previous commit.

vinylspiders committed:
some more qol - when a flashlight is dropped by a mob, it gets dropped facing the same direction as the mob

and another sprite qol pass

vinylspiders committed:
it may be a good idea to use this instead

Helg2 commented:
I don't think this is a good idea to make flashlights change their sprite upon turning without any resprites, because firstly i think they were designed to be one-sided, and secondly it will just look like this:
![image](https://user-images.githubusercontent.com/93882977/229522920-fa0591d6-514e-46b5-bc8e-8d8fa62892b1.png)

vinylspiders committed:
fixes a bug that was causing you to not be able to snuff candles

and also fixes the inhands icon not updating in some cases

vinylspiders committed:
merge conflict

vinylspiders committed:
Merge branch 'master' into robo-eyes

Helg2 commented:
Well, maybe i'm just exaggerating. Can't really compare with in-game experience.

orthography commented:
hehe atomic

vinylspiders committed:
Merge branch 'master' into robo-eyes

vinylspiders committed:
Some refactoring + optimizations, cleanup, more candle lighting qol

Refactors a bit, fixes 'you light the lit flare' message...

Cleans up a bunch of things that were unnecessary.

Some minor optimizations with the process() calls: candle no longer calls update_appearance every tick which is expensive and unnecessary.

Cleans up flare sprites a bit.

vinylspiders committed:
Merge branch 'master' into robo-eyes

vinylspiders committed:
linters, some balloon alerts

vinylspiders committed:
another qol pass for the flare. probably the last.

vinylspiders committed:
balloon alerts made brief

vinylspiders committed:
moved logic to attack chain, final sprite pass

also, I hate the attack chain

vinylspiders committed:
finally taming the beast that is the attack chain

vinylspiders committed:
comment with explanation

vinylspiders committed:
I need to stop now

finally pleased with this.

vinylspiders committed:
one last glanceover

vinylspiders committed:
more descriptive name

vinylspiders committed:
Merge branch 'master' into robo-eyes

vinylspiders committed:
Merge branch 'master' into robo-eyes

vinylspiders committed:
Fixing the merge conflict that I literally brought upon myself

vinylspiders committed:
guh stupid balloon--glad I caught this

vinylspiders committed:
Revert "guh stupid balloon--glad I caught this"

This reverts commit 797711bcc55e32ed7dedf9b7c17bb4a91aa8a462.

vinylspiders committed:
Revert "Fixing the merge conflict that I literally brought upon myself"

This reverts commit 5ec0cce8a0d5d43ff8392bd09128c84ce2878de5.

vinylspiders committed:
you no take candle - revisited

vinylspiders committed:
Revert "you no take candle - revisited"

This reverts commit dca470f9ccf7136b8b1e4d868f1161702a091847.

vinylspiders committed:
confusion resolved

it seems I accidentally included a bunch of the inhand sprites in my other inhand PR. Whoops! They're already merged. I was so confused as to why it wasn't showing items_lefthand and items_righthand in the diffs

vinylspiders committed:
ltubes
